src/mono/mono/metadata/CMakeLists.txt

```diff
@@ -1,5 +1,19 @@
 project(metadata C)
 
+include (${SHARED_CONTAINERS_SOURCE_PATH}containers.cmake)
+
+set(shared_container_sources "")
+
+list(APPEND shared_container_sources
+  ${SHARED_CONTAINER_SOURCES}
+  ${SHARED_CONTAINER_HEADERS}
+)
+
+addprefix(shared_container_sources ${SHARED_CONTAINERS_SOURCE_PATH} "${shared_container_sources}")
+
+add_library(shared_container_objects OBJECT ${shared_container_sources})
+target_include_directories(shared_container_objects PUBLIC ${SHARED_CONTAINERS_SOURCE_PATH})
+
```

Comment:
this should probably be done in `src/mono/mono/CMakeLists.txt`, not in metadata

Reply:
Turns out this isn't a nice to have. it's necessary in order to avoid duplicate symbols in the static linking case.  additionally I need to tackle eventpipe's use of the shared containers in this PR, not separately.